lib/config/configuration_test.go

```diff
@@ -3403,3 +3404,64 @@ func TestApplyDiscoveryConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestApplyFileConfig_DbTokenAuth_errors(t *testing.T) {
+	tests := []struct {
+		name   string
+		url    string
+		scheme string
+		error  string
+	}{
+		{
+			name:   "invalid url template",
+			url:    "{{User}}",
+			scheme: "",
+			error:  "\"User\" not defined",
+		},
+		{
+			name:   "invalid auth scheme",
+			url:    "{{.User}}",
+			scheme: "abcd",
+			error:  "No known authentication scheme abcd",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defaultCfg := service.MakeDefaultConfig()
+			err := ApplyFileConfig(&FileConfig{
+				Databases: Databases{
+					Service: Service{
+						EnabledFlag: "yes",
+					},
+					TokenConfig: TokenConfig{
+						Enabled: types.BoolOption{Value: true},
+						UrlTmpl: test.url,
+						TokenAuthConfig: TokenAuthConfig{
+							Scheme: test.scheme,
+						},
+					},
+				},
+			}, defaultCfg)
+			require.ErrorContains(t, err, test.error)
+		})
+	}
+}
+
+func TestApplyFileConfig_DbTokenAuth(t *testing.T) {
+	defaultCfg := service.MakeDefaultConfig()
+	err := ApplyFileConfig(&FileConfig{
+		Databases: Databases{
```

Comment:
Can add happy test for Full Config (With Optionals like timeout etc) , Minimal Config (Bare minimum required).